types/src/v2/params.rs

```diff
@@ -221,15 +226,34 @@ impl<'a> RpcParamsSequence<'a> {
 	///     seq.optional_next().unwrap(),
 	///     seq.optional_next().unwrap(),
 	///     seq.optional_next().unwrap(),
-	/// ];;
+	/// ];
 	///
 	/// assert_eq!(params, [Some(1), Some(2), None, None]);
 	/// ```
 	pub fn optional_next<T>(&mut self) -> Result<Option<T>, CallError>
 	where
 		T: Deserialize<'a>,
 	{
-		match self.next_inner::<Option<T>>() {
+		match self.next_inner::<Option<T>>(false) {
+			Some(result) => result,
+			None => Ok(None),
+		}
+	}
+
+	/// Similar to `RpcParamsSequence::optional_next` but regards empty JSON Array and JSON Object
```

Comment:
An alternative to this would be to pass the `bool` to `optional_next()` directly, i.e. have the signature `pub fn optional_next<T>(&mut self, empty_means_none: bool) -> Result<Option<T>, CallError>`. Would that work?

Reply:
I don't see a common use case for `optional_next` where the result is an empty json array or json map/object but I kept it to be on the safe-side.

In general I think we should avoid having a public API that has bool flags to do different things, it's so hard to read and easy to get wrong IMHO.